### ISSUE TITLE
Cannot configure multiple paths over same custom port

### ISSUE THREAD

**Issue by herrirya** [aspnet/Diagnostics#514]
It appears it is not possible to have multiple paths that use the same custom port number (which is likely unintentional). For example:

```
app.UseHealthChecks("/health/live", port: 9999);
app.UseHealthChecks("/health/ready", port: 9999);
```

The second configured path in the example ("/health/ready") is not accessible because the first configured path is hogging up the port - seemingly due to the equality condition set up in HealthCheckApplicationBuilderExtensions.cs UseHealthChecksCore() method, line 214.

```
app.MapWhen(
  c => c.Connection.LocalPort == port,
  b0 => b0.Map(path, b1 => b1.UseMiddleware<HealthCheckMiddleware>(args)));
```

This middleware is great; thank you for your work.

**Comment by rynowak** [comment]
Thanks. This looks like this is indeed a bug.

**Comment by rynowak** [comment]
Fixed for the 2.2.0 release. Thanks for the feedback 👍
